Fix de-duplication bug with Veracode SCA

coheigea opened the pull request:
A proposed fix for https://github.com/DefectDojo/django-DefectDojo/issues/6252

StefanFl commented on dojo/tools/veracode/parser.py:
```diff
@@ -202,13 +202,13 @@ def _get_cwe(val):
         finding.static_finding = True
         finding.dynamic_finding = False
-        finding.unique_id_from_tool = xml_node.attrib['cve_id']
+        finding.unique_id_from_tool = dupe_key
```
This is not a good way to use this field. The id here should come directly from the tool and shouldn't be generated. To fix deduplication we should calculate a meaningful hash_code with and use that for deduplication in `settings.dist.py`.

StefanFl commented on unittests/tools/test_veracode_parser_dedup.py:
You shouldn't change the settings in a unit test, it might have effects on other test cases. There must be better ways to achieve this, maybe with a @patch or a mock.

coheigea commented:
@StefanFl @Maffooch Thanks for the review - I updated the PR to just generate a random key for the dupes map in the Veracode parser, and removed the unique_id_from_tool to be the CVE id. This means the de-duplication logic falls back to compute_hash_code_legacy for SCA, which solves the problem.

mtesauro reviewed (approved):
Approved
